pkg/controller/apimanagerrestore/apimanagerrestore_logic_reconciler.go

```diff
@@ -321,6 +330,21 @@ func (r *APIManagerRestoreLogicReconciler) apiManagerFromSharedBackupSecret() (*
 }
 
 func (r *APIManagerRestoreLogicReconciler) reconcileRestoreAPIManager() (reconcile.Result, error) {
+	apiManagerToRestoreName := r.cr.Status.APIManagerToRestoreRef.Name
+	apiManagerAlreadyCreated := true
+
+	err := r.GetResource(types.NamespacedName{Name: apiManagerToRestoreName, Namespace: r.cr.Namespace}, &appsv1alpha1.APIManager{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			apiManagerAlreadyCreated = false
+		}
+		return reconcile.Result{}, err
+	}
+
+	if apiManagerAlreadyCreated {
```

Comment:
this condition is always evaluated to `true`

Reply:
It shouldn't. In line 339 I set it to false :thinking:

Reply:
are you sure? :eyes:

Reply:
:+1: oh I see, I'm always returning when I set it to false. Fixing it...

Reply:
Fixed